### PR TITLE
Test plotly.min.js in Require.js environment

### PR DESCRIPTION
**Pull request by etpinard**
This PR is unfortunately necessary to make sure that `plotly.min.js` can be loaded inside a Jupyter notebook by https://github.com/plotly/plotly.py

See:

- https://github.com/substack/webworkify/issues/29
- https://github.com/plotly/plotly.py/pull/545

for more info.

In brief, this PR:

- add a jasmine test making sure that `plotly.min.js` loads correctly via Require.js
- add a UBER hacky step to the bundle step, where minified bundles pass through a `.replace` to fix the aforementioned issue.

### CODE REVIEW

File: tasks/pretest.js [hunk 1]
```diff
@@ -52,6 +53,16 @@ function makeTestImageFolders() {
     makeOne(constants.pathToTestImagesDiff, 'test image diff folder');
 }
 
+// Make script file that define plotly in a RequireJS context
+function makeRequireJSFixture() {
+    var bundle = fs.readFileSync(constants.pathToPlotlyDistMin, 'utf-8'),
```
Comment (etpinard) [inline tasks/pretest.js]: **Important** I chose to test Require.js import with the `dist/plotly.min.js` bundle only.

That means that Require.js will only be tested on `npm version` commits (when the `dist/` is updated). 

Main reason for this shotcut: minifying the full plotly.js bundle take about 1 minute on CI.

File: tasks/pretest.js [hunk 2]
```diff
@@ -52,6 +53,16 @@ function makeTestImageFolders() {
...
+        template = 'define(\'plotly\', function(require, exports, module) { {{bundle}} });',
+        index = template.replace('{{bundle}}', bundle);
+
+    common.writeFile(constants.pathToRequireJSFixture, index);
```
Comment (etpinard) [inline tasks/pretest.js]: this basically mimics how plotly.py inject `plotly.min.js` - see [here](https://github.com/plotly/plotly.py/blob/22cb9d92bb8485f660aab60cc58ea1791505a097/plotly/offline/offline.py#L147-L157).

File: test/jasmine/karma.conf.js [hunk 3]
```diff
@@ -113,6 +116,13 @@ if(isSingleSuiteRun) {
     func.defaultConfig.preprocessors[pathToMain] = ['browserify'];
     func.defaultConfig.preprocessors[testFileGlob] = ['browserify'];
 }
+else if(isRequireJSTest) {
+    func.defaultConfig.files = [
+        constants.pathToRequireJS,
+        constants.pathToRequireJSFixture,
+        testFileGlob
+    ];
```
Comment (etpinard) [inline test/jasmine/karma.conf.js]: Note: don't browserify the `bundle_tests/requirejs_test.js` suite as browserify and require.js don't get along.

**Comment by bpostlethwaite** [comment]
Yes this is pretty unfortunate but at least you found a way to test it!
:dancer:

**Pull request merged**